intents/simpson_paradox.py

```diff
@@ -0,0 +1,107 @@
+""" 
+	**Some improvement will be done in this code, all ideas for better implementation will be appreciated.
+	1. commenting part will be done later. 
+	2. Implementation part will be improved by using some python function.
+	3. Variable name will be changed according to the need.
+"""
+from Utilities import aspects
+from Utilities import enumerations
+import pandas as pd
+
+def check_simpson_paradox(table, dimension, slice_compare, summary_operator):
+	slice_compare_column = list(slice_compare.keys())[0]
+	dimension.append(slice_compare_column)
+	slice_val_list = list(slice_compare.values())
+	sliced_val = slice_val_list[0]
+	sliced_val.sort()
+
+	table = aspects.group_by(table, dimension, summary_operator)
+
+	num_rows = table.shape[0]
+	num_columns = table.shape[1]
+	positive_count = 0
+	negative_count = 0
+	zero_count = 0
+	total_count = 0
+	row_i = 0
+
+	while row_i < num_rows:
+		if row_i == num_rows - 1:
+			difference = 0
+			if table.loc[row_i][num_columns - 2] == sliced_val[0]:
+				difference = table.loc[row_i][num_columns - 1]
+			else:
+				difference = -table.loc[row_i][num_columns - 1]
+			if difference < 0:
+				negative_count = negative_count + 1
+			elif difference > 0:
+				positive_count = positive_count + 1
+			else:
+				zero_count = zero_count + 1
+			total_count = total_count + 1
+			break
+		equality_check = True
+		for column_i in range(num_columns-2):
+			if table.loc[row_i][column_i] != table.loc[row_i+1][column_i]:
+				equality_check = False
+		if equality_check == True:
+			difference = table.loc[row_i][num_columns - 1] - table.loc[row_i+1][num_columns - 1]
+			if difference < 0:
+				negative_count = negative_count + 1
+			elif difference > 0:
+				positive_count = positive_count + 1
+			else:
+				zero_count = zero_count + 1
+			total_count = total_count + 1
+			row_i = row_i + 1
+		else:
+			difference = 0
+			if table.loc[row_i][num_columns - 2] == sliced_val[0]:
+				difference = table.loc[row_i][num_columns - 1]
+			else:
+				difference = -table.loc[row_i][num_columns - 1]
+			if difference < 0:
+				negative_count = negative_count + 1;
+			elif difference > 0:
+				positive_count = positive_count + 1;
+			else:
+				zero_count = zero_count + 1;
+			total_count = total_count + 1;
+		row_i = row_i + 1
+	return (positive_count/total_count)*100
+    
+def simpson_paradox(table, dimension, grouping, summary_operation_column, slice_compare, summary_operator):
```

Comment:
The function that will be called from another module (essentially be public) should be at the top, while private functions below. Also it is not clear form the name what this function does or returns, does it calculate simpson's paradox and returns a value or just prints a bunch of stuff?